ragaai_catalyst/tracers/utils/rag_trace_json_converter.py

```
@@ -191,55 +266,97 @@ def get_additional_metadata(spans, custom_model_cost, model_cost_dict, prompt=""
+                """Returns the number of tokens in a text string."""
+                try:
+                    encoding = tiktoken.get_encoding(encoding_name)
+                    num_tokens = len(encoding.encode(string))
+                    return num_tokens
+                except Exception as e:
+                    logger.warning(f"Error encoding with {encoding_name}: {str(e)}")
+                    # Fallback to a different encoding if the requested one fails
+                    try:
+                        fallback_encoding = tiktoken.get_encoding("cl100k_base")
+                        return len(fallback_encoding.encode(string))
+                    except:
```

Ruff (0.8.2)

Do not use bare `except`

(E722)